src/lib_ccx/ccx_decoders_708.c

```diff
@@ -1741,10 +1741,10 @@ void ccx_dtvcc_process_current_packet(ccx_dtvcc_ctx *dtvcc)
 				   "Sequence: %d, packet length: %d\n",
 				   seq, len);
 #endif
-	if (dtvcc->current_packet_length != len) // Is this possible?
+	if (dtvcc->current_packet_length != len)
 	{
-		_dtvcc_decoders_reset(dtvcc);
-		return;
+		// Extra padding data, can be ignored
+		dtvcc->current_packet_length = len;
```

Comment:
It's a small thing but worth talking about :-) That variable (current_packet_length) is used here src/lib_ccx/ccx_dtvcc.c by the loop that writes the CC data to the packet buffer... so it originally comes from ourselves.

Is there a bug in the loop? Or some assumption there is incorrect?

On visual inspection I think that if dtvcc->current_packet_length < len then we are in trouble: The encoded data says it has more data than what we actually have, so if we ignore it we're going to get into buffer overruns.

So _maybe_ (this would have to be tested) it would be saner to do 

if (dtvcc->current_packet_length < len) {
// Most likely things are going to be bad for us
len = dtvcc->current_packet_length ; // At least don't read beyond the buffer
}

I'm not sure it's a good idea to overwrite current_packet_length with len. Two reasons
- current_packet_length is the amount data we actually have while len is the amount of data that the file says we should have. So it would be trivial to create a file that makes CCExtractor crash if we trust the external source rather than ourselves :-)
- current_packet_length is not used later in this function at all, so changing it will not affect how data is processed here, but it might affect what happens later.

Reply:
> Is there a bug in the loop? Or some assumption there is incorrect?

As far as i know, it is not a bug in the code. Instead i guess that given file might not follow the specification correctly, as i mentioned in [this comment](https://github.com/CCExtractor/ccextractor/issues/677#issuecomment-796490909) about the two "should be padding" bytes


>So maybe (this would have to be tested) it would be saner to do
if (dtvcc->current_packet_length < len) {
// Most likely things are going to be bad for us
len = dtvcc->current_packet_length ; // At least don't read beyond the buffer
}

I tested it. This causes the extra data to be considered as a service packet, which leads to it being a null service packet header (service number and block size both are 0). We ignore the any packet with service_number=0 in the current code. So i guess we should be fine with this approach


> current_packet_length is not used later in this function at all, so changing it will not affect how data is processed here, but it might affect what happens later.

This shouldn't be an issue because later in the function we clear the packet and set current length to 0


>current_packet_length is the amount data we actually have while len is the amount of data that the file says we should have. So it would be trivial to create a file that makes CCExtractor crash if we trust the external source rather than ourselves :-)

This definitely seems like an issue. We can go ahead with your approach then